modules/account-lib/src/coin/hbar/transferBuilder.ts

```diff
@@ -0,0 +1,116 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
+import Long from 'long';
+import { AccountId } from '@hashgraph/sdk';
+import { proto } from '../../../resources/hbar/protobuf/hedera';
+import { BuildTransactionError, InvalidParameterValueError, SigningError } from '../baseCoin/errors';
+import { BaseKey } from '../baseCoin/iface';
+import { TransactionBuilder, DEFAULT_M } from './transactionBuilder';
+import { Transaction } from './transaction';
+import { isValidAddress, isValidAmount, stringifyAccountId } from './utils';
+
+export class TransferBuilder extends TransactionBuilder {
+  private _txBodyData: proto.CryptoTransferTransactionBody;
+  private _toAddress: string;
+  private _amount: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._txBodyData = new proto.CryptoTransferTransactionBody();
+    this._txBody.cryptoTransfer = this._txBodyData;
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._txBodyData.transfers = this.buildTransferData();
+    this.transaction = await super.buildImplementation();
+    return this.transaction;
+  }
+
+  private buildTransferData(): proto.ITransferList {
+    return {
+      accountAmounts: [
+        { accountID: this.buildAccountData(this._source.address), amount: Long.fromString(this._amount).negate() }, // sender
+        { accountID: this.buildAccountData(this._toAddress), amount: Long.fromString(this._amount) }, // recipient
+      ],
+    };
+  }
+
+  private buildAccountData(address: string): proto.AccountID {
+    const accountData = new AccountId(address);
+    return new proto.AccountID({
+      accountNum: accountData.account,
+      realmNum: accountData.realm,
+      shardNum: accountData.shard,
+    });
+  }
+
+  /** @inheritdoc */
+  protected initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    const transferData = tx.txBody.cryptoTransfer;
+    if (transferData && transferData.transfers && transferData.transfers.accountAmounts) {
+      this.initTransfers(transferData.transfers.accountAmounts);
+    }
+  }
+
+  protected initTransfers(transfers: proto.IAccountAmount[]): void {
+    transfers.forEach(transferData => {
+      const amount = Long.fromValue(transferData.amount!);
+      if (amount.isPositive()) {
+        this.to(stringifyAccountId(transferData.accountID!));
+        this.amount(amount.toString());

```

Comment:
Are these class variables?

Reply:
The ones that we are getting are decoded from the transaction.
The ones that we are setting are the class variables.